src/openrct2/ride/Track.cpp

```diff
@@ -589,7 +589,7 @@ int32_t track_is_connected_by_shape(TileElement* a, TileElement* b)
     return aBank == bBank && aAngle == bAngle;
 }
 
-const rct_preview_track* get_track_def_from_ride(Ride* ride, int32_t trackType)
+const rct_preview_track* get_track_def_from_ride(const Ride* ride, int32_t trackType)
```

Comment:
Not related to this PR, but certainly welcome as another PR. (Even better if that other PR makes a member function out of it.)

Reply:
This change in particular was needed because the ride pointer in TrackDesign::CreateTrackDesignTrack was const, so this function needed to accept a const.

Reply:
You can leave it like this, then.